### PR TITLE
Fix the powers of root of unity

### PR DESCRIPTION
**Pull request by howardwu**
## Motivation

Fix the powers of root of unity. This PR switches the representations into the correct Montgomery form.

### CODE REVIEW

File: curves/src/bls12_377/fq.rs [hunk 1]
```diff
@@ -191,26 +191,34 @@ impl PoseidonDefaultParameters for FqParameters {
...
+        assert_eq!(FqParameters::POWERS_OF_ROOTS_OF_UNITY.len(), powers.len());
 
         // Ensure the expected and candidate powers match.
-        for (expected, candidate) in powers.iter().zip(FqParameters::POWERS_OF_G.iter()) {
-            println!("{:?} =?= {:?}", expected, candidate);
-            assert_eq!(expected, candidate);
+        for (expected, candidate) in powers.iter().zip(FqParameters::POWERS_OF_ROOTS_OF_UNITY.iter()) {
+            // println!("BigInteger({:?}),", expected.0.0);
+            println!("{:?} =?= {:?}", expected.0, candidate);
+            assert_eq!(&expected.0, candidate);
```
Comment (Pratyush) [inline curves/src/bls12_377/fq.rs]:
```suggestion
        for (expected, candidate) in powers.iter().zip(FqParameters::POWERS_OF_ROOTS_OF_UNITY) {
            // println!("BigInteger({:?}),", expected.0.0);
            println!("{:?} =?= {:?}", expected.0, candidate);
            assert_eq!(&expected.0, candidate);
```

File: curves/src/bls12_377/fr.rs [hunk 2]
```diff
@@ -207,26 +207,34 @@ impl PoseidonDefaultParameters for FrParameters {
...
-        assert_eq!(FrParameters::POWERS_OF_G.len(), powers.len());
+        // Ensure the correct number of powers of root of unity are present.
+        assert_eq!(FrParameters::POWERS_OF_ROOTS_OF_UNITY.len() as u64, (FrParameters::TWO_ADICITY - 1) as u64);
+        assert_eq!(FrParameters::POWERS_OF_ROOTS_OF_UNITY.len(), powers.len());
 
         // Ensure the expected and candidate powers match.
-        for (expected, candidate) in powers.iter().zip(FrParameters::POWERS_OF_G.iter()) {
-            println!("{:?} =?= {:?}", expected, candidate);
-            assert_eq!(expected, candidate);
+        for (expected, candidate) in powers.iter().zip(FrParameters::POWERS_OF_ROOTS_OF_UNITY.iter()) {
```
Comment (Pratyush) [inline curves/src/bls12_377/fr.rs]:
```suggestion
        for (expected, candidate) in powers.iter().zip(FrParameters::POWERS_OF_ROOTS_OF_UNITY) {
```

File: curves/src/edwards_bls12/fr.rs [hunk 3]
```diff
@@ -33,7 +33,7 @@ impl Fp256Parameters for FrParameters {}
 impl FftParameters for FrParameters {
     type BigInteger = BigInteger;
 
-    const POWERS_OF_G: &'static [BigInteger] = &[];
+    const POWERS_OF_ROOTS_OF_UNITY: &'static [BigInteger] = unimplemented!();
```
Comment (Pratyush) [inline curves/src/edwards_bls12/fr.rs:36]: Didn't we want to make this an `Option`?

**Review by Pratyush — APPROVED** [review]
LGTM modulo small nits

File: curves/src/edwards_bls12/fr.rs (hunk 3 above)
Comment (howardwu) [reply curves/src/edwards_bls12/fr.rs:36]: Let's revisit this. I think it subtly complicates the `sqrt` logic, and without a fallback `sqrt` algorithm, the `unimplemented!()` likely suffices.

**Pull request merged**